omnigibson/utils/extended_articulation_view.py

```diff
@@ -0,0 +1,234 @@
+from typing import List, Optional, Tuple, Union
+import carb
+import omni.timeline
+from omni.isaac.core.utils.prims import get_prim_at_path
+from pxr import PhysxSchema, Usd, UsdGeom, UsdPhysics
+import numpy as np
+import torch
+import warp as wp
+import math
+
+from omni.isaac.core.articulations import ArticulationView
+
+DEG2RAD = math.pi / 180.0
+
+class ExtendedArticulationView(ArticulationView):
+    """ArticulationView with some additional functionality implemented."""
+
+    def set_joint_limits(
+        self,
+        values: Union[np.ndarray, torch.Tensor],
+        indices: Optional[Union[np.ndarray, List, torch.Tensor, wp.array]] = None,
+        joint_indices: Optional[Union[np.ndarray, List, torch.Tensor, wp.array]] = None,
+    ) -> None:
+        """Sets joint limits for articulation joints in the view.
+
+        Args:
+            values (Union[np.ndarray, torch.Tensor, wp.array]): joint limits for articulations in the view. shape (M, K, 2).
+            indices (Optional[Union[np.ndarray, List, torch.Tensor, wp.array]], optional): indicies to specify which prims
+                                                                                 to manipulate. Shape (M,).
+                                                                                 Where M <= size of the encapsulated prims in the view.
+                                                                                 Defaults to None (i.e: all prims in the view).
+            joint_indices (Optional[Union[np.ndarray, List, torch.Tensor, wp.array]], optional): joint indicies to specify which joints
+                                                                                 to manipulate. Shape (K,).
+                                                                                 Where K <= num of dofs.
+                                                                                 Defaults to None (i.e: all dofs).
+        """
+        if not self._is_initialized:
+            carb.log_warn("ArticulationView needs to be initialized.")
+            return
+        if not omni.timeline.get_timeline_interface().is_stopped() and self._physics_view is not None:
```

Comment:
rubber stamping the functionality -- i presume you copied most of the logic from other functions and then slightly modified it to work with the desired API for joint limits